Makefile

```diff
@@ -35,8 +35,10 @@ TESTFLAGS    :=
 STRESSFLAGS  :=
 DUPLFLAGS    := -t 100
 BUILDMODE    := install
+GOPATH       := $(realpath ../../../..)
```

Comment:
Ah, I don't think it will. Now that you point this out I'm surprised that it ever worked; `go build` must be incorporating the `vendor` directory into the search path for builds invoked in a subdirectory of `vendor`.

This vendoring scenario is a weird use case and I'm not sure if we need to support it in the `Makefile` (if we do support it, it just raises more questions: how does `glock` behave if cockroachdb is vendored?). You can `go build` a vendored copy of cockroachdb and with the changes in #4053 it will work. You don't have to use the `Makefile` (after all this is one of the reasons we go to such lengths to be `go get`-compatible), so I think it's reasonable to require that the `Makefile` only be used from a conventional non-vendored copy of the repo.